### ISSUE TITLE
[3-Code generation] invalid grammar

### ISSUE THREAD

**Issue by BenShen98** [LangProc/langproc-2018-lab#9]
How should we handle the syntax error? Would the behaviour be undefined or do we need to exit the programme with a message like `syntax error`?

Would the testbench test on code that is syntactically correct but has invalid grammar?

**Comment by ymherklotz** [comment]
As far as I know, invalid code will not be tested. The lexer and parser are already written for you as well, so however it handles invalid input is good enough. The only functions you should have to complete are `Compile` and `Interpret`. As there should be no invalid code constructs from the input, it should be fine to just error out.

**Comment by johnwickerson** [comment]
Right, I intend only to test using valid testcases. So you can do whatever you like if you encounter a syntax error.
